h/static/scripts/cross-frame-discovery.coffee

```diff
@@ -0,0 +1,110 @@
+class CrossFrameDiscovery
+  options:
+    # Origins allowed to communicate on the channel
+    origin: '*'
+
+    # When this is true, this bridge will act as a server and, similar to DHCP,
+    # offer to connect to bridges in other frames it discovers.
+    server: false
+
+  constructor: (options) ->
+    @options = $.extend({}, @options, options)
+
+  startDiscovery: (onDiscovery) ->
+    # Find other frames that run the same discovery mechanism. Sends a beacon
+    # and listens for beacons.
+    #
+    # Parameters:
+    # onDiscovery: (source, origin, token) -> ()
+    #   When two frames discover each other, onDiscovery will be called on both
+    #   sides with the same token string.
+    #   <this> will be bound to the CrossFrameDiscovery instance.
+    @onDiscovery = onDiscovery
+
+    # Listen to discovery messages from other frames
+    $(window).on 'message', this._onMessage
+
+    # Send a discovery message to other frames to create channels
+    this._beacon()
+    return
+
+  stopDiscovery: =>
+    # Remove the listener for discovery messages
+    $(window).off 'message', this._onMessage
+    return
+
+
+  # Send out a beacon to discover frames to connect with
+  _beacon: ->
+    beaconMessage = if @options.server
+      '__cross_frame_dhcp_offer'
+    else
+      '__cross_frame_dhcp_discovery'
+
+    # Starting at the top window, walk through all frames, and ping each frame
+    # that is not our own.
+    queue = [window.top]
+    while queue.length
+      parent = queue.shift()
+      if parent isnt window
+        parent.postMessage beaconMessage, @options.origin
+      for child in parent.frames
+        queue.push child
+    return
+
+
+  _onMessage: (e) =>
+    {source, origin, data} = e.originalEvent
+
+    # Fix for local testing (needed at least in Firefox 34.0)
+    if origin is 'null'
+      origin = '*'
```

Comment:
This does not appear in the latest bridge code on master. I'm not familiar with this at all. My guess is that the origin is null if you're testing on a file: URI or something. Kill it.